### PR TITLE
Add github issue and PR templates

### LINKED ISSUE #93: Add GitHub PR templates and hooks
See dicussion at https://3.basecamp.com/3692715/buckets/3007033/messages/507250988

- Add a PR template with a checklist: 
    - docs
    - tests 
    - format

We can add a hook that runs or checks clang-format -- or add the check to travis. Can also have travis fail on compiler warnings.

Also, we can add an issue template (e.g. https://github.com/opencv/opencv/issues/new).

### PR DESCRIPTION
**Pull request by leokoppel**
Resolves #93 

These templates fill the GitHub form fields.

The "Note: enforced by CI check" refers to #104

### CODE REVIEW

**Comment by chutsu** [comment]
Where are the git hooks though? 

Would be useful to have atleast clang-format and run ctest

File: .github/ISSUE_TEMPLATE.md [hunk 1]
```diff
@@ -0,0 +1,17 @@
...
+2.
+
+#### Expected result
+
+#### Observed result
+
+#### Possible cause
+
+#### Related issues
+
```
Comment (chutsu) [inline .github/ISSUE_TEMPLATE.md]: Feels a bit too verbose for me, this issue seems to assume this is for just a "bug" type issue?

File: .github/PULL_REQUEST_TEMPLATE.md [hunk 2]
```diff
@@ -0,0 +1,13 @@
...
+
+<!-- Describe the changes in the pull request. If behaviour has changed,
+describe the previous and new behaviour -->
+
+#### Pre-Merge Checklist:
+- [ ] Code is documented.
+- [ ] Code has automated tests. <!-- If you fixed a bug that was previously
+undetected by tests, it would now be detected. -->
+- [ ] Zero compiler warnings with `-Wall -Wextra`  <!-- Note: enforced by CI check -->
+- [ ] Formatted with `clang-format`                <!-- Note: enforced by CI check -->
```
Comment (chutsu) [inline .github/PULL_REQUEST_TEMPLATE.md]: Are comments necessary?

**Comment by leokoppel** [comment]
This PR only adds templates for the New Issue and New PR forms on github.com. I agree it would be useful to have a git hook as well..

File: .github/ISSUE_TEMPLATE.md (hunk 1 above)
Comment (leokoppel) [inline .github/ISSUE_TEMPLATE.md]: You're right, this assumes it is a bug type issue. It is similar to wavelab/autonomoose#463 

I don't think it is too verbose for a bug type issue. I gave a presentation to WaveLab on this topic: [How to report a bug (if you want it fixed)](https://drive.google.com/open?id=1kLkn-x4H1uTNoWQ-xmQJcRRcd-RyRl3kZl3c9QUfaBQ). The goal is more clear and descriptive bug reports, and such a template could be valuable if we receive bug reports from external users.

For an internal "new feature" type issues we can just clear the form.

File: .github/PULL_REQUEST_TEMPLATE.md (hunk 2 above)
Comment (leokoppel) [inline .github/PULL_REQUEST_TEMPLATE.md]: I guess not. The goal is to give information to the submitter without having it appear in the output.

File: .github/ISSUE_TEMPLATE.md (hunk 1 above)
Comment (chutsu) [inline .github/ISSUE_TEMPLATE.md]: Is there only 1 issue template we can have?

Comment (leokoppel) [inline .github/ISSUE_TEMPLATE.md]: Yes. Gitlab has that feature, github does not.

Comment (chutsu) [inline .github/ISSUE_TEMPLATE.md]: Maybe leave out the top bit and just have a "Pre-Merge Checklist"?

File: .github/PULL_REQUEST_TEMPLATE.md [hunk 3]
```diff
@@ -0,0 +1,9 @@
+Resolves #
+
+
+
+#### Pre-Merge Checklist:
+- [ ] Code is documented in doxygen format
+- [ ] Code has automated tests
+- [ ] Zero compiler warnings
+- [ ] Formatted with `clang-format`
```
Comment (navganti) [inline .github/PULL_REQUEST_TEMPLATE.md]: Don't need clang anymore, since travis fails.

Comment (navganti) [inline .github/PULL_REQUEST_TEMPLATE.md]: Don't need clang anymore, since travis fails.

Comment (leokoppel) [inline .github/PULL_REQUEST_TEMPLATE.md]: Same with compiler warnings, but not everyone knows that.
Good point, on the other hand, If I take it out, I may get messages saying "why is travis failing?"

Comment (navganti) [inline .github/PULL_REQUEST_TEMPLATE.md]: Fair. When you go to the Travis page though on the build, it says that it failed because of clang.

Comment (navganti) [inline .github/PULL_REQUEST_TEMPLATE.md]: It doesn't hurt to have it. This all looks fine to me!

**Pull request merged**